src/changelog/.2.x.x/2331_custom_context_data_in_lookup.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<entry xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns="https://logging.apache.org/xml/ns"
+       xsi:schemaLocation="https://logging.apache.org/xml/ns https://logging.apache.org/xml/ns/log4j-changelog-0.xsd"
+       type="added">
+  <issue id="2331" link="2330_add_faster_web_app_context_map.xml"/>
+  <description format="asciidoc">Add support for custom context data in `ctx` lookup.</description>
```

Comment:
1. Shouldn't this rather be a `fixed`? Yes, we're _adding_ a new behaviour, but one that was missing. Nowhere we had stated that `ctx` lookup (or any particular component) won't work with custom `CDP`s.
2. `issue.link` needs to be fixed.
3. Could you rewrite the `description` a little bit more accurately, please? (Suggestion: _"Fix custom thread-context data provider handling in lookups and filters"_ – since it is not only the `ctx` lookup we fixed.)

Reply:
Fixed in https://github.com/apache/logging-log4j2/pull/2846/commits/c475dbf0af351bc30dc7f94e3f943ce4386ca0fa